kernel/multicore_bringup/src/aarch64.rs

```diff
@@ -50,9 +50,29 @@ pub fn handle_ap_cores(
     ap_data.ap_data_virt_addr.write(VirtualAddress::new_canonical(&ap_data as *const _ as usize));
     ap_data.ap_stage_two_virt_addr.write(VirtualAddress::new_canonical(ap_stage_two as usize));
 
+    // -------
+    //
+    // This code identity-maps one page of writable+executable memory and
+    // copies the machine code of `ap_entry_point` to it. The fact that the
+    // mapping is identity-mapped allows the code to enable the MMU safely,
+    // as PC will be valid before and after the switch.
+
+    // map this RWX
+    let flags = PteFlags::new().valid(true).writable(true).executable(true);
+    let mut ap_startup_mapped_pages = create_identity_mapping(1, flags)?;
+    let virt_addr = ap_startup_mapped_pages.start_address();
+
+    // copy the entry point code
+    // instructions must have an alignment of four bytes; a page's alignment will always be larger.
+    let dst = ap_startup_mapped_pages.as_slice_mut(0, PAGE_SIZE).unwrap();
+    let src = unsafe { (ap_entry_point as *const [u8; PAGE_SIZE]).as_ref() }.unwrap();
```

Comment:
hmm, i think i was able to do this safely on the x86 side, but I'm not 100% sure what differs here in the aarch64 world. Is it possible to map this address and then access it safely via the `MappedPages::as_slice()` API instead of just unsafely trusting that we can dereference it?

Reply:
Sure, I'll use the mod_mgmt lookup

Reply:
Ah yikes, I didn't realize it was that tedious to obtain the backing `MappedPages` object for `ap_entry_point`. There might be a faster way but this is okay for now.